ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -100,7 +100,10 @@ impl<'a> TraceBuilder<'a> {
         // If this unwrap fails, we didn't find the call to the control point and something is
         // profoundly wrong with the AOT IR.
         let trace_inputs = trace_inputs.unwrap();
-        let trace_input_struct_ty = trace_inputs.operand(0).type_(self.aot_mod).as_struct();
+        let trace_input_struct_ty = match trace_inputs.operand(0).type_(self.aot_mod) {
```

Comment:
This is a bit laborious, hence the helper.

Reply:
Yes, but the helper *really* confused me, because it's very unobvious that an `Idx` can do such things.

Reply:
@ptersilie thoughts?

Reply:
Oops, sorry, I confused my helpers! The reason for this one is because `as_struct` bypasses the `enum` in a way that just seems weird.

Reply:
By "bypasses the enum", do you mean, hides the match?

Reply:
Yes: in order to read the code you need to know it's the enum, read the doc of `as_struct` and... that's actually harder than expanding the code here.